driver/ni/ni.h

```diff
@@ -25,448 +25,434 @@
-        pipeline::Acquisition state_write_pipe;
-        bool ok_state = true;
-    };
-
-
-    ///////////////////////////////////////////////////////////////////////////////////
-    //                                    Factory                                    //
-    ///////////////////////////////////////////////////////////////////////////////////
-    class Factory final : public task::Factory{
-    public:
-        std::pair<std::unique_ptr<task::Task>, bool> configureTask( const std::shared_ptr<task::Context> &ctx,
-                                                                    const synnax::Task &task) override;
-
-        std::vector<std::pair<synnax::Task, std::unique_ptr<task::Task> > >
-        configureInitialTasks(      const std::shared_ptr<task::Context> &ctx,
-                                    const synnax::Rack &rack) override;
-
-        ~Factory() = default;
-
-    };
-}
+typedef struct ReaderConfig {
+    std::string device_key;
+    std::vector<ChannelConfig> channels;
+    std::uint64_t sample_rate = 0;
+    std::uint64_t stream_rate = 0;
+    std::string device_name;
+    std::string task_name;
+    std::string timing_source; // for sample clock
+    std::uint64_t period = 0;
+    synnax::ChannelKey task_key;
+    std::set<uint32_t> index_keys;
+} ReaderConfig;
 
+/////////////////////////////////////////////////////////////////////////////////////////////
+//                                                                                         //
+//                                     DAQ INTERFACES                                      //
+//                                                                                         //
+/////////////////////////////////////////////////////////////////////////////////////////////
 
+///////////////////////////////////////////////////////////////////////////////////
+//                                    NiSource                                   //
+///////////////////////////////////////////////////////////////////////////////////
+class Source : public pipeline::Source {
+public:
+    explicit Source(TaskHandle task_handle,
+                    const std::shared_ptr<task::Context> &ctx,
+                    const synnax::Task task);
 
 
-/*
+    int init();
 
-using json = nlohmann::json;
-namespace daq
-{
-    class DaqReader : public pipeline::Source  //TODD: change to daqReader
-    {
-    public:
-        virtual std::pair<synnax::Frame, freighter::Error> read() = 0;
-        virtual freighter::Error start() = 0;
-        virtual freighter::Error stop() = 0;
-        virtual bool ok() = 0;
-    };
+    ~Source();
 
-    class DaqWriter: public pipeline::Sink{
-    public:
-        virtual freighter::Error write(synnax::Frame frame) = 0;
-        virtual freighter::Error start() = 0;
-        virtual freighter::Error stop() = 0;
-        // virtual bool ok() = 0;
-    };
+    void clearTask();
 
+    int checkNIError(int32 error);
```

Comment:
Why do we need such a large amounts of public methods? Confused about how this class reduces complexity